interop-testing/src/main/java/io/grpc/testing/integration/AbstractXdsInteropTest.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2014 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.testing.integration;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.truth.Truth.assertThat;
+import static io.grpc.ConnectivityState.CONNECTING;
+import static io.grpc.ConnectivityState.IDLE;
+import static io.grpc.ConnectivityState.SHUTDOWN;
+import static io.grpc.ConnectivityState.TRANSIENT_FAILURE;
+import static org.junit.Assert.assertEquals;
+
+import io.grpc.ConnectivityState;
+import io.grpc.ConnectivityStateInfo;
+import io.grpc.EquivalentAddressGroup;
+import io.grpc.LoadBalancer;
+import io.grpc.LoadBalancerProvider;
+import io.grpc.Status;
+import io.grpc.xds.orca.OrcaOobUtil;
+import io.grpc.xds.orca.OrcaPerRequestUtil;
+import io.grpc.xds.shaded.com.github.xds.data.orca.v3.OrcaLoadReport;
+import java.util.List;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Abstract base class for all GRPC transport tests.
+ *
+ * <p> New tests should avoid using Mockito to support running on AppEngine.</p>
+ */
+public abstract class AbstractXdsInteropTest extends AbstractInteropTest {
+
+  protected static final String TEST_ORCA_LB_POLICY_NAME = "test_backend_metrics_load_balancer";
+  private final LinkedBlockingQueue<OrcaLoadReport> savedLoadReports = new LinkedBlockingQueue<>();
+  private final LinkedBlockingQueue<OrcaLoadReport> savedOobLoadReports =
+      new LinkedBlockingQueue<>();
+
+  /**
+   *  Test backend metrics reporting: expect the test client LB policy to receive load reports.
+   */
+  public void testOrca() {
+    blockingStub.emptyCall(EMPTY);
+    assertEquals(savedLoadReports.poll(), OrcaLoadReport.newBuilder()
+        .putRequestCost("queue", 2.0).build());
+    assertThat(savedLoadReports.isEmpty()).isTrue();
+    assertEquals(savedOobLoadReports.poll(), OrcaLoadReport.newBuilder()
```

Comment:
This seems racy. I'm considering having the LB delay RPCs until it gets the first load report (or failure). I think we'd need to define a utilization that is guaranteed to have a particular value on server-side.